src/flb_pack.c

```diff
@@ -329,90 +329,90 @@ static int msgpack2json(char* buf, int *off, size_t left, msgpack_object *o)
     case MSGPACK_OBJECT_POSITIVE_INTEGER:
         {
             char temp[32] = {0};
-            snprintf(temp, 31, "%lu", (unsigned long)o->via.u64);
-            ret = try_to_write(buf, off, left,temp,0);
+            ret = try_to_write(buf, off, left, temp,
+                snprintf(temp, 31, "%lu", (unsigned long)o->via.u64));
         }
         break;
 
     case MSGPACK_OBJECT_NEGATIVE_INTEGER:
         {
             char temp[32] = {0};
-            snprintf(temp, 31, "%ld", (signed long)o->via.i64);
-            ret = try_to_write(buf,off,left,temp,0);
+            ret = try_to_write(buf, off, left, temp,
+                 snprintf(temp, 31, "%ld", (signed long)o->via.i64));
```

Comment:
> since you are using temp[32] in different places, declare it for the whole function.

This causes needless allocation in some cases.(e.g     case MSGPACK_OBJECT_BOOLEAN)
So, I use temp[32] only cases which that value is needed.
msgpack_object_print also be implemented in the same way.

> please do not add snprintf() inside try_to_write() as parameter. do it in a separate line, that's fine.

Why? New variable will be needed in your implementation.

> for snprintf buf length parameter, use sizeof(temp) - 1.

I see, I will fix it.